sources/packages/web/src/components/common/students/StudentFileUploads.vue

```diff
@@ -0,0 +1,206 @@
+        :max-width="730"
+        ref="fileUploadModal"
+        title="Upload file"
+        :formData="initialData"
+        formName="uploadStudentDocumentsAEST"
+      >
+        <template #actions="{ cancel, submit }">
+          <v-row class="m-0 p-0">
+            <v-btn color="primary" variant="outlined" @click="cancel"
+              >Cancel</v-btn
+            >
+            <check-permission-role :role="Role.StudentUploadFile">
+              <template #="{ notAllowed }">
+                <v-btn
+                  class="float-right"
+                  @click="submit"
+                  color="primary"
+                  variant="elevated"
+                  :disabled="notAllowed"
+                  >Upload now</v-btn
+                >
+              </template>
+            </check-permission-role>
+          </v-row>
+        </template>
+      </formio-modal-dialog>
+    </body-header-container>
+  </tab-container>
+</template>
+
+<script lang="ts">
+import { onMounted, ref, defineComponent } from "vue";
+import {
+  DEFAULT_PAGE_LIMIT,
+  FormIOForm,
+  PAGINATION_LIST,
+  LayoutTemplates,
+  Role,
+} from "@/types";
+import { StudentService } from "@/services/StudentService";
+import {
+  useFormatters,
+  useFileUtils,
+  ModalDialog,
+  useFormioUtils,
+  useSnackBar,
+} from "@/composables";
+import FormioModalDialog from "@/components/generic/FormioModalDialog.vue";
+import { AESTFileUploadToStudentAPIInDTO } from "@/services/http/dto/Student.dto";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  components: {
+    FormioModalDialog,
+    CheckPermissionRole,
+  },
+  emits: ["loaded"],
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+    canUploadFiles: {
+      type: Boolean,
+      required: true,
+    },
+    canDownloadFiles: {
+      type: Boolean,
+      required: true,
+    },
+    studentFileUploads: {
+      type: Array,
+      required: true,
+    },
+  },
+  setup(props: any, context) {
+    const fileUploadModal = ref({} as ModalDialog<FormIOForm | boolean>);
+    const { dateOnlyLongString } = useFormatters();
+    const snackBar = useSnackBar();
+    const fileUtils = useFileUtils();
+    const formioUtils = useFormioUtils();
+    const initialData = ref({ studentId: props.studentId });
+
+    const loadStudentFileUploads = async () => {
+      context.emit("loaded", props.studentId);
+    };
+
+    onMounted(async () => {
+      await loadStudentFileUploads();
+    });
+
+    const uploadFile = async () => {
+      const modalResult = await fileUploadModal.value.showModal();
+      if (!modalResult) {
+        return;
+      }
+
+      try {
+        const associatedFiles = formioUtils.getAssociatedFiles(modalResult);
+        const payload: AESTFileUploadToStudentAPIInDTO = {
+          associatedFiles,
+        };
+        await StudentService.shared.saveAESTUploadedFilesToStudent(
```

Comment:
@dheepak-aot @andrepestana-aot @dheepak-aot @guru-aot should we move this AEST specific logic outside the component?
Maybe an event that will be triggered with the `associatedFiles` array as an event args?

Reply:
mm... that's a good point...How we are managing in other shared components, I think even though this scenario is AEST specific, it is an action of the component which can be enabled for any client type. I am kind of okay with this.  but would like to know what other devs thought.

Reply:
Thinking about it in a business context, it is only enabled for the ministry for communications. yep, I am good to move it to the AEST side. it makes sense

Reply:
Yes, that's a good idea as it's only related to AEST.

Reply:
I agree

Reply:
I have made the change

Reply:
I am not expecting to provide the modal as a parameter here.
Since the modal itself is in this component, we can call the event if the modal results in a file uploaded. This would be the recommendation from the first comment on this thread where it says "`associatedFiles` array as an event args".